crypto/threads_pthread.c

```diff
@@ -517,10 +523,11 @@ static struct rcu_qp *update_qp(CRYPTO_RCU_LOCK *lock)
         (lock->current_alloc_idx + 1) % lock->group_count;
 
     /* get and insert a new id */
-    new_id = lock->id_ctr;
+    new_id = (uint64_t)lock->id_ctr;
     lock->id_ctr++;
 
     new_id = VAL_ID(new_id);
```

Comment:
Would it be better to combine L526 & L529? or is there reason for this syntax?
```c
new_id = VAL_ID(lock->id_ctr);
```

Reply:
I will look into it now. Can you please printf sizeof: LONG, long, long int on mingw64?
On Win32 VC x86/x64: LONG, long, long int are 32 bit.

Reply:
Looks reasonable, so I changed it. Thanks!